File: OneCodeTeam/Story-Creator-Sample-Application-for-Windows-14638/[C#][C++]-Story Creator Sample Application for Windows Phone 7 (CSWP7AzureVideoStory)/C# and C++/C#,C++/CSWP7AzureVideoStory/VideoStoryCreator/Models/Photo.cs
/****************************** Module Header ******************************\
* Module Name:	Photo.cs
* Project:		VideoStoryCreator
* Copyright (c) Microsoft Corporation.
* 
* The model representing a photo.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System;
using System.IO;
using System.Windows.Media.Imaging;
using VideoStoryCreator.Transitions;

namespace VideoStoryCreator.Models
{
    public class Photo
    {
        public string Name { get; set; }
        public Stream ThumbnailStream { get; set; }
        public WriteableBitmap ResizedImage { get; set; }
        public Stream ResizedImageStream { get; set; }
        public ITransition Transition { get; set; }
        public TimeSpan PhotoDuration { get; set; }

        /// <summary>
        /// We use the name of the photo to do comparison.
        /// </summary>
        public override bool Equals(object obj)
        {
            if (obj is Photo)
            {
                return this.Name.Equals(((Photo)obj).Name);
            }
            return false;
        }

        public override int GetHashCode()
        {
            return this.Name.GetHashCode();
        }
    }
}

File: OneCodeTeam/Story-Creator-Sample-Application-for-Windows-14638/[C#][C++]-Story Creator Sample Application for Windows Phone 7 (CSWP7AzureVideoStory)/C# and C++/C#,C++/CSWP7AzureVideoStory/NativeVideoEncoder/Photo.cpp
/****************************** Module Header ******************************\
* Module Name:	Photo.cpp
* Project: NativeVideoEncoder
* Copyright (c) Microsoft Corporation.
* 
* A model class representing a photo.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

#include "StdAfx.h"
#include "Photo.h"


Photo::Photo(void)
	: m_photoDuration(0),
	m_transitionDuration(0),
	m_pTransition(nullptr)
{
}


Photo::~Photo(void)
{
	if (this->m_pTransition != nullptr)
	{
		delete this->m_pTransition;
		this->m_pTransition = nullptr;
	}
}
